### ISSUE TITLE
Authorizer Facade not working?

### ISSUE THREAD

**Issue by chakkimatti** [lucadegasperi/oauth2-server-laravel#550]
I'm on Laravel 4.2 and I have the correct lines added to the config/app.php as to per L4 installation doc, I have 3.0.2 version of the package installed. 
When I do 
```php
	Route::get('oauth/authorize', ['as' => 'oauth.authorize.get','middleware' => ['check-authorization-params', 'auth'], function ()
    {
	   $authParams = Authorizer::getAuthCodeRequestParams();
	   Log::info(var_dump($authParams ));
	   $formParams = array_except($authParams,'client');
	   $formParams['client_id'] = $authParams['client']->getId();
	   return View::make('auth-form', ['params'=>$formParams,'client'=>$authParams['client']]);
	
    }
```
the Authorizer::getAuthCodeRequestParams() returns an empty array. When doing the query Postman, I get 'undefined index client'.

I'm using grant_type *authorization_code* I've tried doing the query with client, with remember_token as params and without params.

**Comment by vinkla** [comment]
Your issue is very messy, please clean it up before we take a look at it. Always try to write your issues in the finest way for quickest response.

**Comment by chakkimatti** [comment]
Authorizer::getAuthCodeRequestParams(); returns an empty array

**Comment by vinkla** [comment]
How do you register it?

**Comment by chakkimatti** [comment]
I register it just like described on L4 installation document. I got the package by adding it to composer.json and running composer array. The version, I believe is 3.0.2 of the package.

**Comment by vinkla** [comment]
@lucadegasperi any ideas on this?

**Comment by vinkla** [comment]
Could you please cleanup your initial issue and then I can't take a look at that.

**Comment by chakkimatti** [comment]
I'm sorry I can't make it much clearer. The problem seems to be that Authorizer Facade never gets called.

**Comment by chakkimatti** [comment]
Maybe I need some params on the function call? 
    $params = Authorizer::getAuthCodeRequestParams();
If I understand right, the form gets populated with the information about the client. Where does it get the information? Database? Where is that set up?

**Comment by ddelnano** [comment]
If you use the correct markdown it will look cleaner and will be formatted php code.  Read the Github markdown documentation.

**Comment by chakkimatti** [comment]
So, let me rephrase the question.. The 	   $authParams = Authorizer::getAuthCodeRequestParams(); does nothing. Logging the var returns empty. I put some logging inside that method but those never get fired either - and 'check-authorization-params' filter seems to be missing?

**Comment by vinkla** [comment]
Ping @lucadegasperi

**Comment by chakkimatti** [comment]
Should this/is this being called somewhere with a parameter, So far I think the only one is the authCodeRequestDeniedRedirectUri method and that's only for redirect uri and I never get this error anyway. it somehow should check the database for clients and such. Model? Connection configurations?

**Comment by vinkla** [comment]
@chakkimatti did you have any progress with this?

**Comment by chakkimatti** [comment]
No progress :( I'm looking other packages too, but this seems to be the best, built with legendary and all that. How does the  Authorizer::getAuthCodeRequestParams() work? What does it need? What would be the reason it is not called? I tried putting some log entries to the getAuthorizeCodeGrant.php's check-authorization-params Route::filter but it seems it is not called either :(

**Comment by vinkla** [comment]
@lucadegasperi, you can probably give a better answer on this.

**Comment by lucadegasperi** [comment]
Laravel 4.2 doesn't have support for middlewares as I recall right, you should use a route filter instead.

**Comment by vinkla** [comment]
Oh yeah, that is true. Middlewares were introduced in Laravel 5.0 and filters deprecated since 5.1.
